README.md

```diff
@@ -563,12 +569,72 @@ For example
 
 ## Libraries
 
-Most tests will deal with resources and containers (which is a subclass of a resource). These objects are represented
+As previously mentioned, the global variable `clients` holds instances of `SolidClient` which are set up to be able to
+access the test server as a specific user. There are some additional capabilities provided by this class. Most tests
+will deal with resources and containers (which is a subclass of a resource). These objects are represented
 by 2 classes in the test harness: `SolidResouce` and `SolidContainer`. For handling access controls in a universal way
 there are two classes: `AccessDatasetBuilder` and `AccessDataset`. Finally, there are two libraries (`Utils` and
 `RDFModel`) providing parsing functions for headers and RDF respectively. Their functions are exposed globally within
 the Karate environment and have already been described above.
 
+### SolidClient
+The `SolidClient` class provides access to the test server on behalf of a user. This is mostly used internally within
+`SolidResouce` and `SolidContainer` but there are three public API functions available.
+
+#### `getAuthHeaders(method, uri)`
+This returns a map of the authorization headers required for a request on behalf of the user registered in this instance
+of the `SolidClient`. The map can be applied to a request being prepared in Karate using the `headers` keyword: 
+```gherkin
+  And headers clients.alice.getAuthHeaders('GET', url)
+```
+* Parameters:
+  * method - The HTTP method of the request.
+  * url - The absolute url of the request.
+* Returns a map of headers.
+
+#### `send(method, uri, data, headers)` and `sendAuthorized(method, uri, data, headers)`
+This is an alternative way to send a request which allows testers to have full control over the HTTP method and request
+headers. Karate will not allow invalid HTTP methods and sets some headers by default so this method allows some
+additional edge case tests to be performed. The only request header these methods include by default is the `User-Agent`
+so that requests are still traceable to the CTH. There are two versions of this method, the first sends un-authenticated 
+requests, whereas the second adds the correct authorization headers for the user registered in this instance of the
+`SolidClient`.
+* Parameters:
+  * method - The HTTP method of the request.
+  * url - The absolute url of the request.
+  * data - The data to be sent in the request (or null). 
+  * headers - A map of key/value pairs to be send as request headers (or null). See the note below for additional tips.
+* Returns the response as a map with the following structure:
+  ```json5
+  {
+    status: 200, // integer
+    headers: {}, // map of all response headers
+    body: "",    // the response body
+  }
+  ```
+  This allows you to make assertions about the response in the normal manner:
+  ```gherkin
+  Then assert response.status == 201
```

Comment:
It confused me a bit at first that you'd have to say `Then assert response.status` instead of just `Then status`. Now, I think I understand why, but I think it makes sense to actually highlight it rather than say it is a normal, because it is a special case, isn't it?

Reply:
Good point - it is the normal way of doing asserts but NOT the normal way of checking the status. I will amend this.